### PR TITLE
 Remove the need for names in the Dataflow Graph 

### PR DESCRIPTION
**Pull request by Ellpeck**

### CODE REVIEW

File: test/functionality/_helper/dataflow/dataflow-builder-printer.ts [hunk 1]
```diff
@@ -114,7 +116,7 @@ class DataflowBuilderPrinter {
 		}
 		this.recordFnCall(id,'call', [
 			wrap(id),
-			wrap(vertex.name),
+			wrap(this.idMap.get(id)?.lexeme),
```
[the remark this answers is not in the document]
Comment (EagleoutIce) [reply test/functionality/_helper/dataflow/dataflow-builder-printer.ts:119]: Furthermore, with the functions themselves (like `.call`) no longer requiring the name, this argument should be removed!

File: test/functionality/_helper/dataflow/dataflow-builder-printer.ts [hunk 2]
```diff
@@ -46,13 +46,15 @@ const EdgeTypeFnMap: Record<EdgeType, string | undefined> = {
 class DataflowBuilderPrinter {
 	private lines:           Lines = []
 	private graph:           DataflowGraph
+	private idMap:           DataflowMap
 	private rootIds:         Set<NodeId>
 	private coveredVertices: Set<NodeId> = new Set()
 	private coveredEdges:    Set<string> = new Set()
 
-	constructor(graph: DataflowGraph) {
+	constructor(graph: DataflowGraph, idMap: DataflowMap) {
```
Comment (EagleoutIce) [inline test/functionality/_helper/dataflow/dataflow-builder-printer.ts:54]: In general, whenever something besides flowr-core requires an idMap it should work without as well for debugging purposes.

File: test/functionality/dataflow/graph/equal-tests.ts [hunk 3]
```diff
@@ -29,23 +29,22 @@ describe('Equal', () => {
...
+			neq('Additional non-root vertex', emptyGraph(), emptyGraph().use('0', {}, false))
 			neq('Additional edge', emptyGraph(), emptyGraph().reads('0', '1'))
 		})
 		describe('Different elements', () => {
 			describe('Different vertices', () => {
-				const rhs = emptyGraph().use('0', 'x')
-				neq('Id', emptyGraph().use('1', 'x'), rhs)
-				neq('Name', emptyGraph().use('0', 'y'), rhs)
-				neq('Control Dependency', emptyGraph().use('0', 'x', { controlDependencies: ['1'] }), rhs)
+				const rhs = emptyGraph().use('0')
```
Comment (EagleoutIce) [inline test/functionality/dataflow/graph/equal-tests.ts:45]: i am still a little bit torn. now it is borderline impossible to link the implementation to the code without at least generating a mermaid-graph representation. 
This should be definitely addressed within #626. But in general, seeing it this way, i am more of a fan of keeping the names within the builder (we can still forcefully attach a `name` during the construction, and then have the diff checker check if a `name` is a attached and if so, if it is equal to the recovered lexeme - just so we keep the additional sanity check of linking ids to lexemes as well.

**Review by EagleoutIce — CHANGES_REQUESTED** [review]

File: test/functionality/_helper/dataflow/dataflow-builder-printer.ts [hunk 4]
```diff
@@ -46,13 +46,15 @@ const EdgeTypeFnMap: Record<EdgeType, string | undefined> = {
 class DataflowBuilderPrinter {
 	private lines:           Lines = []
 	private graph:           DataflowGraph
+	private idMap?:          DataflowMap
 	private rootIds:         Set<NodeId>
 	private coveredVertices: Set<NodeId> = new Set()
 	private coveredEdges:    Set<string> = new Set()
 
-	constructor(graph: DataflowGraph) {
+	constructor(graph: DataflowGraph, idMap?: DataflowMap) {
```
Comment (EagleoutIce) [inline test/functionality/_helper/dataflow/dataflow-builder-printer.ts:54]: Does the builder no still require the idmap?

Comment (Ellpeck) [reply test/functionality/_helper/dataflow/dataflow-builder-printer.ts:54]: it does not, and eslint neglected to yell at me

**Review by EagleoutIce — APPROVED** [review]

**Comment by EagleoutIce** [comment]
_This pull request is included in v2.0.0 (see [Release v2.0.0 (Dataflow v2)](https://github.com/Code-Inspect/flowr/releases/tag/v2.0.0))._

**Comment by EagleoutIce** [comment]
_This pull request is included in v2.0.0 (see [Release v2.0.0 (Dataflow v2)](https://github.com/Code-Inspect/flowr/releases/tag/v2.0.0))._

**Pull request merged**